jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/TomcatIntegrationTest.java

```diff
@@ -46,67 +46,76 @@ protected JmxScraperContainer customizeScraperContainer(
-                "tomcat.traffic",
-                "The number of bytes transmitted and received",
-                "by",
-                attrs ->
-                    attrs.containsOnly(
-                        entry("proto_handler", "\"http-nio-8080\""), entry("direction", "sent")),
-                attrs ->
-                    attrs.containsOnly(
-                        entry("proto_handler", "\"http-nio-8080\""),
-                        entry("direction", "received"))),
-        metric ->
-            assertGaugeWithAttributes(
-                metric,
-                "tomcat.threads",
-                "The number of threads",
-                "threads",
-                attrs ->
-                    attrs.containsOnly(
-                        entry("proto_handler", "\"http-nio-8080\""), entry("state", "idle")),
-                attrs ->
-                    attrs.containsOnly(
-                        entry("proto_handler", "\"http-nio-8080\""), entry("state", "busy"))),
-        metric ->
-            assertGaugeWithAttributes(
-                metric,
-                "tomcat.max_time",
-                "Maximum time to process a request",
-                "ms",
-                attrs -> attrs.containsOnly(entry("proto_handler", "\"http-nio-8080\""))),
-        metric ->
-            assertSumWithAttributes(
-                metric,
-                "tomcat.request_count",
-                "The total requests",
-                "requests",
-                attrs -> attrs.containsOnly(entry("proto_handler", "\"http-nio-8080\""))));
+  protected MetricsVerifier createMetricsVerifier() {
+    return MetricsVerifier.create()
+        .add(
+            "tomcat.sessions",
+            metric ->
+                metric
+                    .hasDescription("The number of active sessions")
+                    .hasUnit("sessions") // TODO: not aligned with semconv. Should be "{session}"
+                    .isGauge()
+                    .hasDataPointsWithOneAttribute(attributeWithAnyValue("context")))
+        .add(
+            "tomcat.errors",
+            metric ->
+                metric
+                    .hasDescription("The number of errors encountered")
+                    .hasUnit("errors") // TODO: not aligned with semconv. Should be "{error}"
+                    .isCounter()
+                    .hasDataPointsWithOneAttribute(attribute("proto_handler", "\"http-nio-8080\"")))
+        .add(
+            "tomcat.processing_time",
+            metric ->
+                metric
+                    .hasDescription("The total processing time")
+                    .hasUnit("ms")
+                    .isCounter()
+                    .hasDataPointsWithOneAttribute(attribute("proto_handler", "\"http-nio-8080\"")))
+        .add(
+            "tomcat.traffic",
+            metric ->
+                metric
+                    .hasDescription("The number of bytes transmitted and received")
+                    .hasUnit("by")
```

Comment:
```suggestion
                    .hasUnit("By")
```
See https://opentelemetry.io/docs/specs/semconv/general/metrics/#instrument-units https://unitsofmeasure.org/ucum#para-48 and

Reply:
Yes, however the same doubts apply here